### PR TITLE
Improve test suite 2

### PR DESCRIPTION
**Pull request by AlexEvanczuk**
# Summary
This is the first of two PRs to clean up the test suite.

This PR does the following:
1) It takes common test setup that is an implementation detail (and thus distracts from the behavior of the test) into a common helper
2) It cleans up some let blocks
3) It removes an unnecessary, redundant context
4) It adds some test cases for edge cases that are not covered whose behavior might change when I open a PR for https://github.com/BigRails/danger-packwerk/issues/2

The next PR will:
1) Improve `it` block descriptions to be more clear about what the expected outcome is
2) It will create a custom matcher `have_inline_markdown` that encapsulates some common test details, such as:
- It ensures there are no other types of danger messages besides `markdown`
- It identifies what line number the comment is on so its simpler to visualize the danger message

# Commits
- clean up common test setup
- clean up let blocks
- remove unnecessary context
- Add some test cases

### CODE REVIEW

File: spec/danger_packwerk/danger_deprecated_references_yml_changes_spec.rb [hunk 1]
```diff
@@ -42,10 +53,10 @@ module DangerPackwerk
         end
       end
 
-      describe 'behavior when violations are added or removed' do
-        context 'a deprecated_refrences.yml file is added (i.e. a pack has its first violation)' do
-          let(:added_files) do
-            file = write_file('packs/some_pack/deprecated_references.yml', <<~YML.strip)
+      context 'a deprecated_refrences.yml file is added (i.e. a pack has its first violation)' do
```
Comment (technicalpickles) [inline spec/danger_packwerk/danger_deprecated_references_yml_changes_spec.rb:56]:
```suggestion
      context 'a deprecated_references.yml file is added (i.e. a pack has its first violation)' do
```

File: spec/danger_packwerk/danger_deprecated_references_yml_changes_spec.rb [hunk 2]
```diff
@@ -55,126 +66,118 @@ module DangerPackwerk
...
+
+            expect(actual_markdown.message).to eq expected
+            expect(actual_markdown.line).to eq 3
+            expect(actual_markdown.file).to eq 'packs/some_pack/deprecated_references.yml'
+            expect(actual_markdown.type).to eq :markdown
           end
         end
+      end
+
+      context 'a deprecated_refrences.yml file is deleted (i.e. a pack has all violations removed)' do
```
Comment (technicalpickles) [inline spec/danger_packwerk/danger_deprecated_references_yml_changes_spec.rb:144]:
```suggestion
      context 'a deprecated_references.yml file is deleted (i.e. a pack has all violations removed)' do
```

**Review by technicalpickles — APPROVED** [review]
This will make a follow up PR much easier to read, thanks!

Left two suggestions on spelling of the filename, but those were also already there.
